context_test.go

```diff
@@ -0,0 +1,29 @@
+package clickhouse
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestContext(t *testing.T) {
+	// Call context multiple times making sure query options are persisted across calls.
```

Comment:
minor point, but we could achieve it in a way:
```go
t.Run("call context multiple times making sure query options are persisted across calls", func(t *testing.T) {
```
and so on... It gives better visibility on test analysis.

Reply:
Updated, appreciate the quick review.

Reply:
@aaron276h looks good. Added another minor comment.